Firestore/core/src/firebase/firestore/model/transform_operations.h

```diff
@@ -90,6 +98,65 @@ class ServerTimestampTransform : public TransformOperation {
   }
 };
 
+// TODO(mikelehen): ArrayTransform can only be used from Obj-C until we switch
+// to using FieldValue instead of FSTFieldValue.
+#if defined(__OBJC__)
+/**
+ * Transforms an array via a union or remove operation (for convenience, we use
+ * this class for both Type::ArrayUnion and Type::ArrayRemove).
+ */
+class ArrayTransform : public TransformOperation {
+ public:
+  ArrayTransform(const Type& type, const std::vector<FSTFieldValue*> elements)
+      : type_(type), elements_(elements) {
+  }
+
+  ~ArrayTransform() override {
+  }
+
+  Type type() const override {
+    return type_;
+  }
+
+  std::vector<FSTFieldValue*> elements() const {
```

Comment:
As is this returns a copy of the elements. Could this return a const reference?

```c++
const std::vector<FSTFieldValue*>& elements() const {
  return elements_;
}
```

If the caller wants to change the array they can make their own copy.

Reply:
Good idea, done.